crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -1606,6 +1713,14 @@ impl<'a> Checker<'a> {
                 .ancestor_ids(self.semantic.scope_id)
                 .find_or_last(|scope_id| !self.semantic.scopes[*scope_id].kind.is_generator())
                 .unwrap_or(self.semantic.scope_id)
+        } else if flags.contains(BindingFlags::CLASS_ATTRIBUTE)
```

Comment:
I'm not sure if this is the best way to do this. I would rather to pass the scope to this function. Because the scope where the attribute should be saved is calculated once before this function is called. However changing the function signature would be a larger refactor for this rule.